node/bft/src/gateway.rs

```diff
@@ -274,12 +274,16 @@ impl<N: Network> Gateway<N> {
     /// Returns `true` if the given IP is this node.
     pub fn is_local_ip(&self, ip: SocketAddr) -> bool {
         ip == self.local_ip()
-            || (ip.ip().is_unspecified() || ip.ip().is_loopback()) && ip.port() == self.local_ip().port()
+            || (ip.ip().is_unspecified() || ip.ip().is_loopback() && ip.ip() == self.local_ip().ip())
+                && ip.port() == self.local_ip().port()
```

Comment:
I think it would be a good idea to break this logic down into smaller chunks, and to comment on them for extra clarity